app/src/main/java/com/graeberj/taskman/auth/presentation/login/LoginScreen.kt

```diff
@@ -19,77 +17,77 @@ import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.SpanStyle
-import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.text.buildAnnotatedString
-import androidx.compose.ui.text.font.Font
-import androidx.compose.ui.text.font.FontFamily
-import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.withStyle
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.graeberj.taskman.R
 import com.graeberj.taskman.auth.presentation.components.CustomButton
-import com.graeberj.taskman.auth.presentation.components.CustomTextField
-import com.graeberj.taskman.ui.theme.Black
+import com.graeberj.taskman.auth.presentation.components.EmailTextField
+import com.graeberj.taskman.auth.presentation.components.PasswordTextField
+import com.graeberj.taskman.auth.presentation.components.TopGreeting
 
 @Composable
-fun LoginScreen() {
-    Column(modifier = Modifier.fillMaxSize().background(Black)) {
+fun LoginScreen(
+    onSignupClick: ()-> Unit
+) {
+    val viewModel: LoginViewModel = hiltViewModel()
+    val state = viewModel.state
+
```

Comment:
I'd avoid passing the ViewModel directly to screen composables. This has 2 major reasons:
1. You're making it harder to have a preview, since you'll need to provide a fake ViewModel instance and fakes for everything the ViewModel needs in its constructor.
2. UI tests are also a bit harder since you'll need to provide these fakes.

What I suggest instead:
```
// In your ViewModel
@HiltViewModel
class SampleViewModel @Inject constructor(): ViewModel() {
    private val _state = MutableStateFlow(SampleState())
    val state = _state.asStateFlow()

    fun onEvent(event: SampleEvent) {
        when(event) {
            is SampleEvent.Event1 -> // Handle event1
            is SampleEvent.Event2 -> // Handle event2
            // ...
        }
    }

    // ...
}

// In your NavHost
composable(Routes.sample) {
    val viewModel = hiltViewModel<SampleViewModel>()
    val state by viewModel.state.collectAsState()
    SampleScreen(
        state = state,
        onEvent = viewModel::onEvent
    )
}

// Your screen composable
@Composable
fun SampleScreen(
    state: SampleState,
    onEvent: (SampleEvent) -> Unit
) {
    // ...
    Button(onClick = { onEvent(SampleEvent.ClickedButtonX) } {
        Text("Click me")
    }
}
```